EssentialFeed/EssentialFeediOSTests/ImageCommentsSnapshotTests.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  ImageCommentsSnapshotTests.swift
+//  EssentialFeediOSTests
+//
+//  Created by Adrian Szymanowski on 16/03/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import XCTest
+import EssentialFeediOS
+@testable import EssentialFeed
+
+class ImageCommentsSnapshotTests: XCTestCase {
+	
+	func test_emptyList() {
+		let sut = makeSUT()
+		
+		sut.display(emptyList())
+		
+		assert(snapshot: sut.snapshot(for: .iPhone8(style: .light)), named: "EMPTY_IMAGE_COMMENTS_LIGHT")
+		assert(snapshot: sut.snapshot(for: .iPhone8(style: .dark)), named: "EMPTY_IMAGE_COMMENTS_DARK")
+	}
+	
+	func test_loadingState() {
+		let sut = makeSUT()
+
+		sut.refreshControl?.beginRefreshing()
+
+		assert(snapshot: sut.snapshot(for: .iPhone8(style: .light)), named: "IMAGE_COMMENTS_LOADING_LIGHT")
+		assert(snapshot: sut.snapshot(for: .iPhone8(style: .dark)), named: "IMAGE_COMMENTS_LOADING_DARK")
+	}
+	
+	func test_commentsWithContent() {
+		let sut = makeSUT()
+		
+		sut.display(commentsWithContent())
+		
+		assert(snapshot: sut.snapshot(for: .iPhone8(style: .light)), named: "IMAGE_COMMENTS_WITH_CONTENT_LIGHT")
+		assert(snapshot: sut.snapshot(for: .iPhone8(style: .dark)), named: "IMAGE_COMMENTS_WITH_CONTENT_DARK")
+		assert(snapshot: sut.snapshot(for: .iPhone8(style: .dark, contentSize: .extraExtraLarge)), named: "IMAGE_COMMENTS_WITH_CONTENT_DARK_EXTRA_EXTRA_LARGE")
+	}
+	
+	func test_commentsWithErrorMessage() {
+		let sut = makeSUT()
+		
+		sut.display(ImageCommentsErrorViewModel(message: "This is a\nmulti-line\nerror message"))
+
+		assert(snapshot: sut.snapshot(for: .iPhone8(style: .light)), named: "IMAGE_COMMENTS_WITH_ERROR_MESSAGE_LIGHT")
+		assert(snapshot: sut.snapshot(for: .iPhone8(style: .dark)), named: "IMAGE_COMMENTS_WITH_ERROR_MESSAGE_DARK")
+		assert(snapshot: sut.snapshot(for: .iPhone8(style: .dark, contentSize: .extraExtraLarge)), named: "IMAGE_COMMENTS_WITH_ERROR_MESSAGE_DARK_EXTRA_EXTRA_LARGE")
+	}
+	
+	// MARK: - Helpers
+	
+	private func makeSUT() -> ImageCommentsViewController {
+		let bundle = Bundle(for: FeedViewController.self)
+		let storyboard = UIStoryboard(name: "ImageComments", bundle: bundle)
+		let controller = storyboard.instantiateInitialViewController() as! ImageCommentsViewController
+		controller.loadViewIfNeeded()
+		controller.tableView.showsVerticalScrollIndicator = false
+		controller.tableView.showsHorizontalScrollIndicator = false
+		return controller
+	}
+	
+	private func emptyList() -> [ImageCommentCellController] {
+		[]
+	}
+	
+	private func commentsWithContent() -> [ImageCommentCellController] {
+		[
+			ImageCommentViewModel(
+				id: UUID().uuidString,
+				authorUsername: "Ἀριστοτέλης",
+				createdAt: "2020-10-09T11:24:59+0000",
```

Comment:
Use relative strings in the test data to mimic the real production code.

```suggestion
				createdAt: "1 day ago",
```